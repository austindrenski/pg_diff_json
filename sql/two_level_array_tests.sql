-- two-level array tests

\echo 'two-level array test (identity)'
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$);

\echo 'two-level array test (delete)'
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0,
  "b": [
    0,
    1
  ]
}$$);

\echo 'two-level array test (full delete)'
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0
}$$);

\echo 'two-level array test (new value)'
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0,
  "b": [
    0,
    1,
    3
  ]
}$$);

\echo 'two-level array test (new type)'
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0,
  "b": [
    0,
    1,
    false
  ]
}$$);

\echo 'two-level test (new value and new type, multilevel)'
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 1,
  "b": [
    0,
    1,
    false
  ]
}$$);