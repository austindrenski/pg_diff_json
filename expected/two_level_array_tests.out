-- two-level array tests

\echo 'two-level array test (identity)'
two-level array test (identity)
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$);
 diff_jsonb 
------------
 {}
(1 row)


\echo 'two-level array test (delete)'
two-level array test (delete)
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0,
  "b": [
    0,
    1
  ]
}$$);
     diff_jsonb     
--------------------
 {"(-,\"{b,2}\",)"}
(1 row)


\echo 'two-level array test (full delete)'
two-level array test (full delete)
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0
}$$);
  diff_jsonb  
--------------
 {"(-,{b},)"}
(1 row)


\echo 'two-level array test (new value)'
two-level array test (new value)
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0,
  "b": [
    0,
    1,
    3
  ]
}$$);
     diff_jsonb      
---------------------
 {"(+,\"{b,2}\",3)"}
(1 row)


\echo 'two-level array test (new type)'
two-level array test (new type)
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 0,
  "b": [
    0,
    1,
    false
  ]
}$$);
       diff_jsonb        
-------------------------
 {"(+,\"{b,2}\",false)"}
(1 row)


\echo 'two-level test (new value and new type, multilevel)'
two-level test (new value and new type, multilevel)
SELECT diff_jsonb($${
  "a": 0,
  "b": [
    0,
    1,
    2
  ]
}$$, $${
  "a": 1,
  "b": [
    0,
    1,
    false
  ]
}$$);
             diff_jsonb              
-------------------------------------
 {"(+,\"{b,2}\",false)","(+,{a},1)"}
(1 row)

